commit 21be79782e8623cad4120253fdc221a19ea63211
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Feb 1 14:21:47 2020 +0100

    Remove MS_LAZYTIME mount flag
    
    The sys/mount.h header in glibc-2.17 (CentOS 7) doesn't know
    about MS_LAZYTIME yet.

diff --git a/lxd/seccomp/seccomp.go b/lxd/seccomp/seccomp.go
index b9695b792..910672ceb 100644
--- a/lxd/seccomp/seccomp.go
+++ b/lxd/seccomp/seccomp.go
@@ -310,7 +310,7 @@ move_mount errno 38
 //                          MS_NOEXEC | MS_SYNCHRONOUS | MS_MANDLOCK |
 //                          MS_DIRSYNC | MS_NOATIME | MS_NODIRATIME | MS_REC |
 //                          MS_VERBOSE | MS_SILENT | MS_POSIXACL | MS_RELATIME |
-//                          MS_STRICTATIME | MS_LAZYTIME;
+//                          MS_STRICTATIME;
 //
 // Now we inverse the flag:
 //
@@ -1191,7 +1191,7 @@ type MountArgs struct {
 	shift  bool
 }
 
-const knownFlags C.ulong = C.MS_BIND | C.MS_LAZYTIME | C.MS_MANDLOCK |
+const knownFlags C.ulong = C.MS_BIND | C.MS_MANDLOCK |
 	C.MS_NOATIME | C.MS_NODEV | C.MS_NODIRATIME |
 	C.MS_NOEXEC | C.MS_NOSUID | C.MS_REMOUNT |
 	C.MS_RDONLY | C.MS_STRICTATIME |
@@ -1201,7 +1201,6 @@ const knownFlagsRecursive C.ulong = knownFlags | C.MS_REC
 var mountFlagsToOptMap = map[C.ulong]string{
 	C.MS_BIND:            "bind",
 	C.ulong(0):           "defaults",
-	C.MS_LAZYTIME:        "lazytime",
 	C.MS_MANDLOCK:        "mand",
 	C.MS_NOATIME:         "noatime",
 	C.MS_NODEV:           "nodev",
diff --git a/lxd/seccomp/seccomp_test.go b/lxd/seccomp/seccomp_test.go
index 6d46dd797..fff6ecdf1 100644
--- a/lxd/seccomp/seccomp_test.go
+++ b/lxd/seccomp/seccomp_test.go
@@ -10,12 +10,12 @@ import (
 
 func TestMountFlagsToOpts(t *testing.T) {
 	opts := mountFlagsToOpts(knownFlags)
-	if opts != "ro,nosuid,nodev,noexec,sync,remount,mand,noatime,nodiratime,bind,strictatime,lazytime" {
+	if opts != "ro,nosuid,nodev,noexec,sync,remount,mand,noatime,nodiratime,bind,strictatime" {
 		t.Fatal(fmt.Errorf("Mount options parsing failed with invalid option string: %s", opts))
 	}
 
 	opts = mountFlagsToOpts(knownFlagsRecursive)
-	if opts != "ro,nosuid,nodev,noexec,sync,remount,mand,noatime,nodiratime,rbind,strictatime,lazytime" {
+	if opts != "ro,nosuid,nodev,noexec,sync,remount,mand,noatime,nodiratime,rbind,strictatime" {
 		t.Fatal(fmt.Errorf("Mount options parsing failed with invalid option string: %s", opts))
 	}
 }
diff --git a/lxd/storage/drivers/utils.go b/lxd/storage/drivers/utils.go
index 0cc9aed22..596d54a48 100644
--- a/lxd/storage/drivers/utils.go
+++ b/lxd/storage/drivers/utils.go
@@ -367,7 +367,6 @@ var mountOptions = map[string]mountOption{
 	"diratime":      {false, unix.MS_NODIRATIME},
 	"dirsync":       {true, unix.MS_DIRSYNC},
 	"exec":          {false, unix.MS_NOEXEC},
-	"lazytime":      {true, unix.MS_LAZYTIME},
 	"mand":          {true, unix.MS_MANDLOCK},
 	"noatime":       {true, unix.MS_NOATIME},
 	"nodev":         {true, unix.MS_NODEV},
diff --git a/lxd/storage_utils.go b/lxd/storage_utils.go
index 5a37a6b9e..35c2aa8bf 100644
--- a/lxd/storage_utils.go
+++ b/lxd/storage_utils.go
@@ -101,7 +101,6 @@ var mountOptions = map[string]mountOption{
 	"diratime":      {false, unix.MS_NODIRATIME},
 	"dirsync":       {true, unix.MS_DIRSYNC},
 	"exec":          {false, unix.MS_NOEXEC},
-	"lazytime":      {true, unix.MS_LAZYTIME},
 	"mand":          {true, unix.MS_MANDLOCK},
 	"noatime":       {true, unix.MS_NOATIME},
 	"nodev":         {true, unix.MS_NODEV},
